packages/components/src/form/Form.tsx

```diff
@@ -0,0 +1,130 @@
+import { DevTool } from '@hookform/devtools'
+import { createContext, useContext } from 'react'
+import {
+  useFormContext,
+  FormProvider,
+  useForm,
+  useFormState,
+  FieldValues,
+} from 'react-hook-form'
+import { FormTitle, Panel } from './'
+
+//*****************************************************************************
+// Interface
+//*****************************************************************************
+
+export interface Props {
+  id?: string;
+  mode?: 'onBlur' | 'onChange' | 'onSubmit' | 'onTouched' | 'all';
+  onSubmit?: (data: any, event: any) => void;
+  onError?: () => void;
+  onDirtyChange?: () => void;
+  defaultValues?: { [key: string]: any };
+  title?: string;
+  withPanel?: boolean;
+  panelClassName?: string; // applied to panel
+  formClassName?: string; // applied to form element
```

Comment:
@ianmarquez should we merged this two props as customStyles for the Form component? What do you think? and also maybe include a root, for the outer wrapper.